### ISSUE TITLE
Null exception thrown in CreateInlineSchema

### ISSUE THREAD

**Issue by SecretDeveloper** [domaindrivendev/Swashbuckle#194]
Hi,

I am adding Swashbuckle to a 1.0 web api project.  When I browse to /swagger I can see that the page loads but the call to load /swagger/docs/v1 fails with a 500 error status.

Digging around some more in the stack trace of the exception being thrown I get the following:

```
{"Message":"An error has occurred.","ExceptionMessage":"One or more errors occurred.","ExceptionType":"System.AggregateException","StackTrace":"   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at My.Web.API.APIMessageLogger.<SendAsync>b__1(Task`1 task) in c:\\Projects\\Merges\\Development\\Release\\Web\\My.Web.API\\API\\APIMessageLogger.cs:line 30
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at System.Web.Http.HttpServer.<SendAsync>d__0.MoveNext()","InnerException":{"Message":"An error has occurred.","ExceptionMessage":"Object reference not set to an instance of an object.","ExceptionType":"System.NullReferenceException","StackTrace":"   
   at Swashbuckle.Swagger.SchemaRegistry.CreateInlineSchema(Type type, String refPrefix)
   at Swashbuckle.Swagger.SchemaRegistry.GetOrRegister(Type type)
   at Swashbuckle.Swagger.SwaggerGenerator.CreateParameter(ApiParameterDescription paramDesc, Boolean inPath, SchemaRegistry schemaRegistry)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Swashbuckle.Swagger.SwaggerGenerator.CreateOperation(ApiDescription apiDescription, SchemaRegistry schemaRegistry)
   at Swashbuckle.Swagger.SwaggerGenerator.CreatePathItem(IEnumerable`1 apiDescriptions, SchemaRegistry schemaRegistry)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Swashbuckle.Swagger.SwaggerGenerator.GetSwagger(String rootUrl, String apiVersion)
   at Swashbuckle.Application.SwaggerDocsHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpMessageInvoker.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Web.Http.Dispatcher.HttpRoutingDispatcher.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Web.Http.Cors.CorsMessageHandler.<SendAsync>d__0.MoveNext()"}}
```

Looking at the code on https://github.com/domaindrivendev/Swashbuckle/blob/master/Swashbuckle.Core/Swagger/SchemaRegistry.cs 

EDIT: After some further digging I can see it is the _jsonContractResolver that is null.  Any idea why that would be the case?

Thanks,
Gary

**Comment by SecretDeveloper** [comment]
This looks like the same issue seen in https://github.com/domaindrivendev/Swashbuckle/issues/186

**Comment by SecretDeveloper** [comment]
Ok so after even more digging around it appears to happen when the api has been registered with some additional JsonContractFormatters in its Application_Start function.

I have the following configuration:

``` csharp
JsonSerializerSettings jSettings = new Newtonsoft.Json.JsonSerializerSettings();
GlobalConfiguration.Configuration.Formatters.JsonFormatter.SerializerSettings = jSettings;
```
which is used to remove the k__BackingField properties in our models when they are serialized.  If I remove this code swagger seems to work ok.

I think if the above configuration is included then the  ContractResolver is set to null which causes HttpConfigurationExtensions.GetJsonContractResolver to set_jsonContractResolver to null which would cause the entire problem.

I was able to resolve the problem by including the following line in my Application_Start:
``` csharp
jSettings.ContractResolver = new JsonContractResolver(new JsonMediaTypeFormatter());
```

**Comment by amanzoor** [comment]
Thanks Gary. that works, Interestingly enough 4.0 did not require this additional configuration on app start. I hope we can get this resolved so that we don't have to do this.

**Comment by domaindrivendev** [comment]
Should be fixed in latest release (5.0.1)
